Generators/EqualityAnalyzer.cs

```diff
@@ -46,7 +44,7 @@ public override void Initialize(AnalysisContext context)
             context.RegisterCompilationStartAction(context =>
             {
                 var compilation = context.Compilation;
-                var typeIEquatableT = compilation.GetTypeByMetadataName(typeof(IEquatable<>).FullName!) as INamedTypeSymbol;
+                INamedTypeSymbol? typeIEquatableT = compilation.GetTypeByMetadataName(typeof(IEquatable<>).FullName!);
```

Comment:
`GetTypeByMetadataName` already returns INamedTypeSymbol, the cast was redundant.